### ISSUE TITLE
False-positive "x doesn't implement `Display` (required by {})"

### ISSUE THREAD

**Issue by s-titov** [intellij-rust/intellij-rust#10508]
## Environment

* **IntelliJ Rust plugin version:** 0.4.194.5382-231
* **Rust toolchain version:** rustc 1.69.0 (84c898d65 2023-04-16)
* **IDE name and version:** Goland 2023.1.2
* **Operating system:** Windows 11

## Problem description
I've implemented fmt::Display for Point2D and it works, but IDE still show me error.

<img width="517" alt="image" src="https://github.com/intellij-rust/intellij-rust/assets/41398445/f4d8ae87-448a-4aca-af7b-cde444879cbf">

## Steps to reproduce

```rust
use std::fmt;

struct Point2D {
    x: f64,
    y: f64,
}

impl fmt::Display for Point2D {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "x: {}, y: {}", self.x, self.y)
    }
}

fn main() {
    let point = Point2D { x: 3.3, y: 7.2 };
    println!("Display: {}", point);
}
```

**Comment by asorgejr** [comment]
I have observed this consistently happening when I add a local crate as a dependency. Everything compiles fine, however the IDE only recognizes the type and not the implementation of `Display`.
- Rustc / cargo: 1.72.0
- Rust Plugin: 0.4.200.5421-232
- CLion 2023.2.1
- Manjaro Linux
